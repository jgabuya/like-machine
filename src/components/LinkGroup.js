import React from "react";
import PropTypes from "prop-types";
import LinkItem from "./LinkItem";

import { sortLinksByLikeCountDesc } from "../util/data-operations";
import moment from "moment";
import { confirmAlert } from "react-confirm-alert";
import normalizeUrl from "normalize-url";

import "react-confirm-alert/src/react-confirm-alert.css";

const propTypes = {
    linkGroup: PropTypes.object.isRequired,
    user: PropTypes.any,
    onDelete: PropTypes.func.isRequired,
    onLike: PropTypes.func.isRequired,
    onUnlike: PropTypes.func.isRequired
};

const renderLinks = props => {
    return props.links.map((item, index) => {
        return (
            <LinkItem
                key={index}
                id={item.id}
                title={item.title}
                description={item.description}
                url={normalizeUrl(item.url)}
                imageUrl={item.image_url}
                createdAt={item.created_at}
                liked={item.liked}
                likeCount={item.like_count}
                owned={item.owned}
                user={props.user}
                onLike={() => {
                    props.onLike(item.id);
                }}
                onUnlike={() => {
                    props.onUnlike(item.id);
                }}
                onDelete={() => {
                    confirmAlert({
                        title: "Confirm delete",
                        message: "Are you sure?",
                        buttons: [
                            {
                                label: "Yes",
                                onClick: () => props.onDelete(item.id)
                            },
                            {
                                label: "No"
                            }
                        ]
                    });
                }}
            />
        );
    });
};

const LinkGroup = props => {
    return (
        <div>
            <h4 className="mb-3 text-muted">
                {moment.unix(props.linkGroup.timestamp).format("MMMM D, YYYY")}
            </h4>

            {renderLinks({
                links: sortLinksByLikeCountDesc(props.linkGroup.links),
                user: props.user,
                onLike: props.onLike,
                onUnlike: props.onUnlike,
                onDelete: props.onDelete
            })}
        </div>
    );
};

LinkGroup.propTypes = propTypes;

export default LinkGroup;