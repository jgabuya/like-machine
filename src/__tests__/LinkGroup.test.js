import React from "react";
import LinkGroup from "../components/LinkGroup";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import moment from 'moment';

describe("LinkGroup", () => {
    const props = {
        linkGroup: { timestamp: moment().unix(), links: [] },
        user: null,
        onDelete: jest.fn(),
        onLike: jest.fn(),
        onUnlike: jest.fn()
    };

    const wrapper = shallow(<LinkGroup {...props} />);

    it("matches snapshot", () => {
        const tree = renderer.create(<LinkGroup {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders without crashing", () => {
        expect(wrapper.length).toEqual(1);
    });
});
