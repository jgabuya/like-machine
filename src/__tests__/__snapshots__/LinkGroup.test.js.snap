// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkGroup matches snapshot 1`] = `
<div>
  <h4
    className="mb-3 text-muted"
  >
    September 30, 2018
  </h4>
</div>
`;
